presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1591,7 +1605,7 @@ public Expression rewriteIdentifier(Identifier reference, Void context, Expressi
                 List<List<Set<Expression>>> enumeratedGroupingSets = node.getGroupBy().get().getGroupingElements().stream()
                         .map(GroupingElement::enumerateGroupingSets)
                         .collect(toImmutableList());
-
+                checkGroupingElementCount(node.getGroupBy().get(), enumeratedGroupingSets);
```

Comment:
By this point, it's already too late. A single big CUBE could generate a huge number of entries in `enumeratedGroupingSets`. This needs to happen before `GroupingElement::enumerateGroupingSets` is called above.